src/riscv_machine.c

```diff
@@ -68,13 +84,18 @@ typedef struct RISCVMachine {
 #define CLINT_SIZE      0x000c0000
 #define HTIF_BASE_ADDR 0x40008000
 #define IDE_BASE_ADDR  0x40009000
+#define UART_BASE_ADDR 0x4000A000
+#define UART_IRQ       1
 #define VIRTIO_BASE_ADDR 0x40010000
 #define VIRTIO_SIZE      0x1000
-#define VIRTIO_IRQ       1
+#define VIRTIO_IRQ       2
 #define PLIC_BASE_ADDR 0x40100000
 #define PLIC_SIZE      0x00400000
 #define FRAMEBUFFER_BASE_ADDR 0x41000000
 
+
+/***************************   RTC   ***************************/
```

Comment:
Can you split section headers like this that are not related to the UART implementation onto separate commit? One extra commit that says something like "Insert comments to mark the device specific code section" would do.